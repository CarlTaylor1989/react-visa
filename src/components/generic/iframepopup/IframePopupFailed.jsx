import React from 'react';
import { Translate } from 'react-redux-i18n';

import './IframePopupFailed.scss';

const IframePopupFailed = () => (
  <div className="iframePopupFailed">
    <span className="icon" />
    <div className="errorBody">
      <Translate className="errorTitle" value="generic.iframePopup.errorTitle" />
      <Translate className="errorText" value="generic.iframePopup.errorText" />
    </div>
  </div>
);

export default IframePopupFailed;
