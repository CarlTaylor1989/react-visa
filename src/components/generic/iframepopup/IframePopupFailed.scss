@import '../../../styles/variables';
@import '../../../styles/mixins';

$error-icon: '../../../styles/icons/iframepopup/error_icon.svg';
$error-icon-width: 125px;
$error-icon-height: 125px;

.iframePopupFailed {
  @include border-radius(24px);
  border: 2px solid $colour-white;
  display: flex;
  flex-direction: row;
  left: 50%;
  padding: 40px;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  z-index: 2;

  .icon {
    background-image: url($error-icon);
    background-position: center center;
    background-repeat: no-repeat;
    background-size: $error-icon-width $error-icon-height;
    display: block;
    height: $error-icon-height;
    width: $error-icon-width;
  }

  .errorBody {
    margin: 30px 0 30px 40px;

    .errorTitle {
      color: $colour-gold;
      display: block;
      font-size: 48px;
      margin-bottom: 20px;
    }

    .errorText {
      color: $colour-white;
      font-size: 36px;
    }
  }
}
