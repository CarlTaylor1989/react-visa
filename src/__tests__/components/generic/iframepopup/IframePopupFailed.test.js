/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import IframePopupFailed from '../../../../components/generic/iframepopup/IframePopupFailed';

test('renders the IframePopupFailed component', () => {
  const wrapper = shallow(<IframePopupFailed />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
