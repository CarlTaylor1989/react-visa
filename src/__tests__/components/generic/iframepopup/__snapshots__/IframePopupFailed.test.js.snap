// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the IframePopupFailed component 1`] = `
<div
  className="iframePopupFailed"
>
  <span
    className="icon"
  />
  <div
    className="errorBody"
  >
    <Translate
      className="errorTitle"
      tag="span"
      value="generic.iframePopup.errorTitle"
    />
    <Translate
      className="errorText"
      tag="span"
      value="generic.iframePopup.errorText"
    />
  </div>
</div>
`;
